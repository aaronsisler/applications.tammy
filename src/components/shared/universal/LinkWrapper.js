import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class LinkWrapper extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="link_wrapper">
                <Link
                    className="nav_link"
                    onClick={this.props.onClick}
                    to={this.props.to}
                >
                    {this.props.linkText}
                </Link>
            </div>
        );
    }
}

LinkWrapper.propTypes = {
    linkText: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    to: PropTypes.string.isRequired,
};

