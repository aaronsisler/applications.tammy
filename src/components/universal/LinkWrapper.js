import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class LinkWrapper extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <Link
                className="link_wrapper nav_link"
                onClick={this.props.onClick}
                to={this.props.to}
            >
                {this.props.linkText}
            </Link>
        );
    }
}

LinkWrapper.propTypes = {
    linkText: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    to: PropTypes.string.isRequired,
};
