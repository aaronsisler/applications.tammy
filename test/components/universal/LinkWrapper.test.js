import React from 'react';
import { shallow } from 'enzyme';
import LinkWrapper from 'Universal/LinkWrapper';

describe('LinkWrapper', () => {
    const mockLinkText = 'mock link text';
    const onClickMock = jest.fn();
    const to = 'to location';
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <LinkWrapper
                linkText={mockLinkText}
                onClick={onClickMock}
                to={to}
            />
        );
    })

    it('should render LinkWrapper correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call the onClick prop when clicked', () => {
        wrapper.find('.link_wrapper').simulate('click');

        expect(onClickMock).toHaveBeenCalled();
    });
});
