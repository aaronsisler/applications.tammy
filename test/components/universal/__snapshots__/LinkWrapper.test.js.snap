// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkWrapper should render LinkWrapper correctly 1`] = `
<Link
  className="link_wrapper nav_link"
  onClick={[MockFunction]}
  replace={false}
  to="to location"
>
  mock link text
</Link>
`;
